accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -6,6 +6,14 @@ Status: Draft
 
 ## CHANGELOG
 
+2020.12.30
+  - Remove the warning for overrides with different default values.
+  - Specify canonicalization of type literals.
+  - Specify the unsound mode legacy rewrite of type arguments to const
+    constructors and literals.
+  - Specify that constant object canonicalization is done with respect to normal
+    forms of generic type arguments.
```

Comment:
I think I would probably propose to just do this as a breaking change.  It won't affect anyone.  We have already made this breaking change for non const runtime type objects, by the way.

Reply:
Ah yes, we've had that behavior since `dart-sdk-2.8.0-dev.17.0`, even a 100% legacy program will use normalization during canonicalization. I hadn't thought about that. So it was a breaking change that didn't actually break anything. Amazing, there are many lines of Dart code out there! ;-)

In that case I certainly think it's reasonable to keep using normalization.

We might even be in a situation where normalization cannot be observed in very many other cases, and that might allow us to use normalization more aggressively. We might then be able to think "normalization occurs whenever possible" rather than "normalization occurs only in a few specific situations".

We may then be able to apply some specialized optimizations. Perhaps each normalized run-time type representation will make subtype checks faster, perhaps other optimizations will be allowed only when _every_ type is normalized.

Nice!

Reply:
(Could resolve this thread, but I'll keep it open for now because it seems important to be aware of this fact.)